src/test/Cardano/CoinSelectionSpec.hs

```diff
@@ -139,8 +145,35 @@ spec = do
     lowerConfidence :: Confidence
     lowerConfidence = Confidence (10^(6 :: Integer)) 0.75
 
+-- A collection of general properties that should apply to all algorithms
+-- that implement the 'CoinSelectionAlgorithm' interface.
+--
+coinSelectionAlgorithmGeneralProperties
```

Comment:
why not to use `Common` or `Generic` ? it sounds better to me (I know you know better as you are native speaker :-) )

Reply:
> why not to use Common or Generic ? it sounds better to me (I know you know better as you are native speaker :-) )

Good question!

I'm trying to capture a contrast between _specialized_ and _generalized_.

We already have tests for properties that are specific (or specialized) to particular algorithms. For example, for `Largest-First`, we have a specialized property test that checks that all selected inputs are greater than or equal (in coin value) to the remaining inputs not selected.

However, the properties in this PR are all general (non-specific), as they apply to all implementations.

We could also use the term "common" here. I tried to avoid the word "generic" as it could be confused with the concept of generic programming.